### PR TITLE
Fix an edge case scenario where an error doesn't have a backtrace

### PR DESCRIPTION
**Pull request by moltenice**

### CODE REVIEW

File: lib/twiglet/logger.rb [hunk 1]
```diff
@@ -40,12 +40,14 @@ def warning(message)
...
-                                  'error': {
-                                    'message': error.message,
-                                    'stack_trace': error.backtrace.join("\n")
-                                  }
-                                })
+        message = message.merge(
+          {
+            'error': {
+              'message': error.message,
+              'stack_trace': (error.backtrace.join("\n") if error.backtrace)
```
Comment (sleepyfox) [inline lib/twiglet/logger.rb]:
```suggestion
              'stack_trace': error.backtrace ? error.backtrace.join("\n") : ""
```
and ditch the compact. Also, I'd be suspicious of any error that doesn't have a stacktrace...

File: test/logger_test.rb [hunk 2]
```diff
@@ -182,12 +182,23 @@
...
+    assert_equal 'Connection timed-out', actual_log[:error][:message]
+    assert_nil actual_log[:error][:stack_trace]
+  end
+
   levels = [
-    { method: :debug, level: 'debug' },
-    { method: :info, level: 'info' },
-    { method: :warning, level: 'warning' },
-    { method: :warn, level: 'warning' },
-    { method: :error, level: 'error' }
```
Comment (sleepyfox) [inline test/logger_test.rb]: I think we should fix Rubucop so that it allows the more readable Hashes, YMMV.

File: lib/twiglet/logger.rb [hunk 3]
```diff
@@ -40,12 +40,14 @@ def warning(message)
...
     def error(message, error = nil)
       if error
-        message = message.merge({
-                                  'error': {
-                                    'message': error.message,
-                                    'stack_trace': error.backtrace.join("\n")
-                                  }
-                                })
+        message = message.merge(
+          {
```
Comment (sleepyfox) [inline lib/twiglet/logger.rb]: I'd move the opening brace onto the line above as I think it's more readable. YMMV.

**Review by sleepyfox — APPROVED** [review]
Minor comments.

File: lib/twiglet/logger.rb (hunk 1 above)
Comment (sleepyfox) [reply lib/twiglet/logger.rb]: Actually a better approach might be to add a .addStackTrace method to our Hash class.

File: lib/twiglet/logger.rb (hunk 3 above)
Comment (moltenice) [reply lib/twiglet/logger.rb]: This moves everything too much to the right (see how it was before)! So there's this huge empty area and the code is all the way to the right for no reason and to me breaks the flow when reading the code :)

File: test/logger_test.rb (hunk 2 above)
Comment (moltenice) [reply test/logger_test.rb]: Perhaps as a separate card? Should be straightforward and we can re-format everything in that PR.

File: lib/twiglet/logger.rb (hunk 1 above)
Comment (moltenice) [reply lib/twiglet/logger.rb]: The problem with the suggested is that it has an empty stack_trace value. The compact removes this value so it doesn't exist in the final JSON.

Comment (sleepyfox) [reply lib/twiglet/logger.rb]: I think you're misinterpreting what I've said, I'm suggesting that we create a .addStackTrace method that we call, that will add the attribute if the stack trace is present and do nothing if it is absent.

Comment (moltenice) [reply lib/twiglet/logger.rb]: Aha I like that!

Comment (moltenice) [reply lib/twiglet/logger.rb]: Done!

File: lib/twiglet/logger.rb (hunk 3 above)
Comment (sleepyfox) [reply lib/twiglet/logger.rb]: YMMV

**Pull request merged**